spec/rubocop/cop/style/to_json_spec.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Style::ToJson, :config do
+  let(:config) { RuboCop::Config.new('Layout/IndentationWidth' => { 'Width' => 2 }) }
+
+  it 'registers an offense when using JSON.generate' do
+    expect_offense(<<~RUBY)
+      JSON.generate(foo)
+      ^^^^^^^^^^^^^^^^^^ Use `.to_json` instead of `JSON.generate`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      foo.to_json
+    RUBY
+  end
+
+  it 'correctly autocorrects calls with implicit hashes' do
+    expect_offense(<<~RUBY)
+      JSON.generate(hello: :world)
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `.to_json` instead of `JSON.generate`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      { hello: :world }.to_json
+    RUBY
+  end
+
+  it 'correctly autocorrects calls with serialization options' do
+    expect_offense(<<~RUBY)
+      JSON.generate(foo, space_before: '  ', space_after: ' ')
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `.to_json` instead of `JSON.generate`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      foo.to_json(space_before: '  ', space_after: ' ')
+    RUBY
+  end
```

Comment:
I think we'll want to add specs for some or all of the following non-offending edge cases where we don't understand the arguments:

- `JSON.generate` (no args)
- `JSON.generate(one, two, three)`
- `JSON.generate(one, two, three: true)`

As well as `JSON.generate(one, two)` where `two` is not a `Hash` literal. e.g.
```ruby
options = ...
JSON.generate(blah, options)
```